pkg/apis/config/types.go

```diff
@@ -78,6 +78,8 @@ type MetricProviderSpec struct {
 	Address string
 	// The authentication token of the metric provider
 	Token string
+	// Whether to enable the InsureSkipVerify options for https requests on Metric Providers.
+	InsecureSkipVerify string
```

Comment:
@wangchen615 you will need to add it in the versioned types.go (pkg/apis/config/v1beat3/types.go), and use codegen utilities to cover the conversion, defaulting, etc.

Reply:
@Huang-Wei , yes. Sorry, I was just pulling updates for a test and mistakenly did a merge. Will do a rebase and re-add my changes.

Reply:
Hi, @zorro786 , thanks for the pointer. Yeah, I actually copied my repo to `$GOPATH/src/sigs.k8s.io/` to run `./hack/update-codegen.sh` but it does not really change the generated code. I think it is because we did not change the names of those set default functions. For example: https://github.com/wangchen615/scheduler-plugins/blob/5bed368b02c1e06a975ccd46cdc9e15de2f5982b/pkg/apis/config/v1beta3/zz_generated.defaults.go#L40 remains the same regardless of whether we adding a new `if` condition within `SetDefaultLoadVariationRiskBalancingArgs `

Reply:
`zz_generated.defaults.go` will not be generated, need to manually add code there.
For other codegen files, try `make clean` first and then run the script.

Reply:
Regarding zz_generated.defaults.go, its format is documented at at https://github.com/kubernetes/code-generator/blob/8d7923dd2451259fb5fc16bc9fbaeaca0b255506/cmd/defaulter-gen/main.go#L19-L21. Say all SetDefaults_Type function will be auto-wired into RegisterDefaults(), and RegisterDefaults() will be placed in register.go.

However, somehow some PR renames  SetDefaults_Type to SetDefaultsType, which makes defaulter-gen doesn't pick up new functions automatically. We're working on a new PR fixing this.

But in any case, the defaulting logic itself needs to be composed manually.